pandas/tests/io/formats/test_console.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+from functools import partial
+
+from pandas.io.formats.console import detect_console_encoding, locale
+
+
+def mock_raises_exception(error=Exception):
+    raise error
+
+
+@pytest.mark.parametrize('empty,filled', [
+    ['stdin', 'stdout'],
+    ['stdout', 'stdin']
+])
+def test_detect_console_encoding_from_stdout_stdin(monkeypatch, empty, filled):
+    # Ensures that when sys.stdout.encoding or sys.stdin.encoding is used when
+    # they have values filled.
+    class MockEncoding(object):
+        def __init__(self, encoding):
+            super(MockEncoding, self).__init__()
+            self.encoding = encoding
+
+    monkeypatch.setattr('sys.{}'.format(empty), MockEncoding(''))
```

Comment:
Any chance you saw the context manager for monkeypatch? Seems to be preferred for patching standard library items (may require a version bump for pytest dependency).